src/components/shared/FormInfo.js

```diff
@@ -32,54 +40,103 @@ const FormInfo = (props) => {
     nameButton,
   } = props;
 
-  const initialState = useMemo(
-    () => ({
-      title: '',
-      release_date: '',
-      poster_path: '',
-      genres: '',
-      overview: '',
-      runtime: '',
-    }),
-    []
-  );
+  const initialValues = {
+    id: '',
+    title: '',
+    date: '',
+    url: '',
+    genre: '',
+    overview: '',
+    runtime: '',
+  };
 
-  const [selectedOption, setSelectedOption] = useState(null);
+  const FormSchema = Yup.object({
+    id: Yup.number().integer('Invalid format'),
+    title: Yup.string().required('Required'),
+    date: Yup.date().required('Required'),
+    url: Yup.string().url('Invalid format').required('Required'),
+    genre: Yup.string().nullable().required('Required'),
+    overview: Yup.string().required('Required'),
+    runtime: Yup.number()
+      .integer('Invalid format')
+      .positive('Invalid format')
+      .required('Required'),
+  });
 
-  const setSelectValue = useCallback(() => {
-    const { genres } = filmEdit;
+  const updateFilms = (films, editFilm) => {
```

Comment:
I suppose this function acts like a helper and doesn't depend on component state.
In that case, this should be moved into action's helpers.
More than that, this function is mutating the passed object `films`

Reply:
fix